app/lib/features/chat/widgets/image_message_builder.dart

```diff
@@ -1,101 +1,138 @@
+            child: Image.memory(
+              decodedImage!,
+              errorBuilder: (
+                BuildContext context,
+                Object url,
+                StackTrace? error,
+              ) {
+                return Text('Could not load image due to $error');
+              },
+              frameBuilder: (
+                BuildContext context,
+                Widget child,
+                int? frame,
+                bool wasSynchronouslyLoaded,
+              ) {
+                if (wasSynchronouslyLoaded) {
+                  return child;
+                }
+                return AnimatedOpacity(
+                  opacity: frame == null ? 0 : 1,
+                  duration: const Duration(seconds: 1),
+                  curve: Curves.easeOut,
+                  child: child,
+                );
+              },
+              fit: BoxFit.contain,
+            ),
           ),
         );
       }
-    } else if (message.uri.isNotEmpty && isURL(message.uri)) {
-      // remote url
       return ClipRRect(
         borderRadius: BorderRadius.circular(15),
-        child: CachedNetworkImage(
-          imageUrl: message.uri,
-          width: messageWidth.toDouble(),
-          errorWidget: (
+        child: Image.memory(
+          decodedImage!,
+          errorBuilder: (
             BuildContext context,
             Object url,
-            dynamic error,
+            StackTrace? error,
           ) {
             return Text('Could not load image due to $error');
           },
+          frameBuilder: (
+            BuildContext context,
+            Widget child,
+            int? frame,
+            bool wasSynchronouslyLoaded,
+          ) {
+            if (wasSynchronouslyLoaded) {
+              return child;
+            }
+            return AnimatedOpacity(
+              opacity: frame == null ? 0 : 1,
+              duration: const Duration(seconds: 1),
+              curve: Curves.easeOut,
+              child: child,
+            );
+          },
+          width: widget.messageWidth.toDouble(),
+          fit: BoxFit.cover,
         ),
       );
-    }
-    // local path
-    // the image that just sent is displayed from local not remote
-    else {
+    } else if (widget.message.uri.isNotEmpty && isURL(widget.message.uri)) {
+      // remote url
       return ClipRRect(
         borderRadius: BorderRadius.circular(15),
-        child: Image.file(
-          File(message.uri),
-          width: messageWidth.toDouble(),
-          errorBuilder: (
+        child: CachedNetworkImage(
+          imageUrl: widget.message.uri,
+          width: widget.messageWidth.toDouble(),
+          errorWidget: (
             BuildContext context,
-            Object error,
-            StackTrace? stackTrace,
+            Object url,
+            dynamic error,
           ) {
             return Text('Could not load image due to $error');
           },
         ),
```

Comment:
Are we sure this works in encrypted rooms? Because there the data at the given uri is scrambled and unreadable unless you know how to decrypt it ... I think we need to return back to either room or client and hand that the uri to download and decrypt it for  us ...

(not sure we want to tackle that in this PR, but probably want to take note of it, if not)

Reply:
Technically, it would never reach to this for now as we're fetching binaries for all images of room. The uri will always be empty. This is just added as support in case if we plan to use uri rather than binaries for remote images.